gateway-server/src/main/java/org/apache/knox/gateway/services/security/impl/ZookeeperRemoteAliasService.java

```diff
@@ -44,461 +44,403 @@
+                    }
+                }
+            }
         }
-      }
     }
-  }
-
-  /**
-   * Check to make sure all the required entries are properly set up
-   */
-  private static void checkPathsExist(
-      final RemoteConfigurationRegistryClient remoteClient) {
-    ensureEntry(PATH_KNOX, remoteClient);
-    ensureEntry(PATH_KNOX_SECURITY, remoteClient);
-    ensureEntry(PATH_KNOX_ALIAS_STORE_TOPOLOGY, remoteClient);
-    ensureEntry(
-        PATH_KNOX_ALIAS_STORE_TOPOLOGY + PATH_SEPARATOR + NO_CLUSTER_NAME,
-        remoteClient);
-  }
-
-  /**
-   * Get a list of all aliases for a given cluster.
-   * Remote aliases are preferred over local.
-   *
-   * @param clusterName cluster name
-   * @return List of all the aliases
-   */
-  @Override
-  public List<String> getAliasesForCluster(final String clusterName) throws AliasServiceException {
-    List<String> remoteAliases = new ArrayList<>();
-
-    /* If we have remote registry configured, query it */
-    if (remoteClient != null) {
-      remoteAliases = remoteClient
-          .listChildEntries(buildClusterEntryName(clusterName));
+
+    /**
+     * Check to make sure all the required entries are properly set up
+     */
+    private static void checkPathsExist(final RemoteConfigurationRegistryClient remoteClient) {
+        ensureEntry(PATH_KNOX, remoteClient);
+        ensureEntry(PATH_KNOX_SECURITY, remoteClient);
+        ensureEntry(PATH_KNOX_ALIAS_STORE_TOPOLOGY, remoteClient);
+        ensureEntry(PATH_KNOX_ALIAS_STORE_TOPOLOGY + PATH_SEPARATOR + NO_CLUSTER_NAME, remoteClient);
     }
 
-    return remoteAliases;
-  }
-
-  @Override
-  public void addAliasForCluster(final String clusterName,
-      final String alias, final String value)
-      throws AliasServiceException {
-    if (remoteClient != null) {
-      final String aliasEntryPath = buildAliasEntryName(clusterName, alias);
-
-      /* Ensure the entries are properly set up */
-      checkPathsExist(remoteClient);
-      ensureEntry(buildClusterEntryName(clusterName), remoteClient);
-      try {
-        remoteClient.createEntry(aliasEntryPath, encrypt(value));
-      } catch (Exception e) {
-        throw new AliasServiceException(e);
-      }
-
-      if (remoteClient.getEntryData(aliasEntryPath) == null) {
-        throw new IllegalStateException(String.format(Locale.ROOT,
-            "Failed to store alias %s for cluster %s in remote registry", alias,
-            clusterName));
-      }
+    /**
+     * Get a list of all aliases for a given cluster. Remote aliases are preferred over local.
+     *
+     * @param clusterName
+     *            cluster name
+     * @return List of all the aliases
+     */
+    @Override
+    public List<String> getAliasesForCluster(final String clusterName) throws AliasServiceException {
+        return remoteClient == null ? new ArrayList<>() : remoteClient.listChildEntries(buildClusterEntryName(clusterName));
```

Comment:
So, this method doesn't check the local alias service at all?

Reply:
This is correct; I was only focusing on retrieving passwords as quickly as possible. Let me fix this and submit a new patch soon.

Reply:
Fixed.